internal/controllers/app_controller.go

```diff
@@ -173,6 +173,29 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App) reconci
 			message: err.Error(),
 		}
 	}
+
+	// check for canary deployment
+	if app.Status.IsActiveCanary {
+		if app.Status.CurrentCanaryStep != app.Spec.Canary.Steps {
+			if *app.Spec.Canary.NextScheduledTime == metav1.Now() {
+				// update traffic weight distributions across deployments
+				app.Spec.Deployments[0].RoutingSettings.Weight = app.Spec.Deployments[0].RoutingSettings.Weight - app.Spec.Canary.StepWeight
+				app.Spec.Deployments[1].RoutingSettings.Weight = app.Spec.Deployments[1].RoutingSettings.Weight + app.Spec.Canary.StepWeight
+
+				// check if the canary weight is exceeding 100% of traffic
+				if app.Spec.Deployments[1].RoutingSettings.Weight >= 100 {
```

Comment:
one thing that we have in `shipa` is that we scale pod count as well. 
If an app has 100 pods, you probably want 100 steps to look like:
step 1:   v1 - 99 pods and 99% of traffic, v2 - 1 pod and 1% of traffic
step 2:   v1 - 98 pods and 98% of traffic, v2 - 2 pods and 2% of traffic
step 3:   v1 - 97 pods and 97% of traffic, v2 - 3 pods and 3% of traffic
....